src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/ODataServiceDocumentResourceV2Provider.cs

```diff
@@ -102,7 +101,12 @@ public ODataServiceDocumentResourceV2Provider()
             try
             {
                 lastRequestUri = await client.ProcessResponseAsync(
-                    () => new HttpRequestMessage(HttpMethod.Get, url),
+                    () =>
+                    {
+                        var request = new HttpRequestMessage(HttpMethod.Get, url);
+                        request.SetLogger(log);
+                        return request;
```

Comment:
suggestion: since this code is used multiple times, and request.setlogger is called after every instantiation, can we create a factory method that take log, url and httpmethod as parameters and returns an instance of HttpRequestMessage with the logger set?